stdlib/typing.pyi

```diff
@@ -631,6 +631,7 @@ class Mapping(Collection[_KT], Generic[_KT, _VT_co]):
     def keys(self) -> KeysView[_KT]: ...
     def values(self) -> ValuesView[_VT_co]: ...
     def __contains__(self, __key: object) -> bool: ...
+    def __eq__(self, other: object) -> bool: ...
```

Comment:
We have lots of classes in typeshed that we pretend inherit from `Mapping` in the stubs, even though they don't at runtime. This is because `Mapping` is an ABC not a protocol, so it's the only way to get type checkers to recognise these classes as being mappings.

A consequence of this is that a bunch of classes that fake-inherit from `Mapping` in the stubs will only allow the second argument to be passed as a positional argument. So I think we should mark it as positional-only in the stubs, even though that's not actually true at runtime, to reduce false negatives for those classes.

We already do a similar thing for lots of other methods in `typing.pyi` — these classes are very special in the typing system!

```suggestion
    def __eq__(self, __other: object) -> bool: ...
```